src/frequenz/dispatch/_actors_service.py

```diff
@@ -0,0 +1,225 @@
+
+        @classmethod
+        def new_with_dispatch(
+                cls,
+                initial_dispatch: DispatchUpdate,
+                dispatch_updates_receiver: Receiver[DispatchUpdate],
+                *,
+                name: str | None = None,
+        ) -> "Self":
+            self = cls(name=name)
+            self._dispatch_updates_receiver = dispatch_updates_receiver
+            self._update_dispatch_information(initial_dispatch)
+            return self
+
+        @override
+        async def _run(self) -> None:
+            other_recv: Receiver[Any] = ...
+
+            if self._dispatch_updates_receiver is None:
+                async for msg in other_recv:
+                    # do stuff
+                    ...
+            else:
+                await self._run_with_dispatch(other_recv)
+
+        async def _run_with_dispatch(self, other_recv: Receiver[Any]) -> None:
+            async for selected in select(self._dispatch_updates_receiver, other_recv):
+                if selected_from(selected, self._dispatch_updates_receiver):
+                    self._update_dispatch_information(selected.message)
+                elif selected_from(selected, other_recv):
+                    # do stuff
+                    ...
+                else:
+                    assert False, f"Unexpected selected receiver: {selected}"
+
+        def _update_dispatch_information(self, dispatch_update: DispatchUpdate) -> None:
+            print("Received update:", dispatch_update)
+            self._dry_run = dispatch_update.dry_run
+            self._options = dispatch_update.options
+            match dispatch_update.components:
+                case []:
+                    print("Dispatch: Using all components")
+                case list() as ids if isinstance(ids[0], int):
+                    component_ids = ids
+                case [ComponentCategory.BATTERY, *_]:
+                    component_category = ComponentCategory.BATTERY
+                case unsupported:
+                    print(
+                        "Dispatch: Requested an unsupported selector %r, "
+                        "but only component IDs or category BATTERY are supported.",
+                        unsupported,
+                    )
+
+    async def main():
+        url = os.getenv("DISPATCH_API_URL", "grpc://fz-0004.frequenz.io:50051")
+        key  = os.getenv("DISPATCH_API_KEY", "some-key")
+
+        microgrid_id = 1
+
+        dispatcher = Dispatcher(
+            microgrid_id=microgrid_id,
+            server_url=url,
+            key=key
+        )
+        dispatcher.start()
+
+        status_receiver = dispatcher.new_running_state_event_receiver("EXAMPLE_TYPE")
+
+        def actor_factory(initial_dispatch, dispatch_receiver):
+            return MyActor.new_with_dispatch(initial_dispatch, dispatch_receiver)
+
+        managing_actor = DispatchManagingActor(
+            actor_factory=actor_factory,
```

Comment:
In this particular case you can just use:
```suggestion
            actor_factory=MyActor.new_with_dispatch,
```